aten/src/ATen/native/xnnpack/Factory.cpp

```diff
@@ -35,17 +34,31 @@ Tensor empty_with_tail_padding(
   return tensor.resize_(size, memory_format);
 }
 
-Tensor allocate_padded_if_needed(const Tensor& input_contig) {
-  const auto* allocator = input_contig.storage().allocator();
-  const auto* guarding_allocator = get_guarding_allocator();
-  if (allocator == guarding_allocator) {
-    return input_contig;
+Tensor allocate_padded_contiguous_if_needed(
+    const Tensor& input,
+    const c10::MemoryFormat memory_format) {
+  const auto* const allocator = input.storage().allocator();
+  const auto* const guarding_allocator = get_guarding_allocator();
+
+  // If the allocators are the same and the memory is contiguous in the requested
+  // format, then there is no need to reallocate the tensor.
+
+  if ((allocator == guarding_allocator) && input.is_contiguous(memory_format)) {
```

Comment:
This function is called only with `MemoryFormat::ChannelsLast`, right? If there is no other use case, then I am wondering if we should make that explicit in the function name and not pass memory_format argument? Just a thought.